cpp/include/cudf/strings/strings_column_handler.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <vector>
+
+#include <rmm/thrust_rmm_allocator.h>
+#include <cudf/column/column_view.hpp>
+#include <cudf/column/column.hpp>
+
+namespace cudf {
+
+/**---------------------------------------------------------------------------*
+ * @brief Given a column-view of strings type, an instance of this class
+ * provides the strings operations on the column.
+ *---------------------------------------------------------------------------**/
+class strings_column_handler
+{
+ public:
+  ~strings_column_handler() = default;
+
+  strings_column_handler( const column_view& strings_column, rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource() );
+  //strings_column_handler( const column_view&& strings_column );
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the number of strings in the column
+   *---------------------------------------------------------------------------**/
+  size_type count() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a pointer to the internal char data array
+   *---------------------------------------------------------------------------**/
+  const char* chars_data() const;
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a pointer to the internal offsets array
+   *---------------------------------------------------------------------------**/
+  const int32_t* offsets_data() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the size of the char data array in bytes
+   *---------------------------------------------------------------------------**/
+  size_type chars_column_size() const;
```

Comment:
I think just return the `column_view` to the `char`s child and then you can query the `size` directly there.